### PR TITLE
Test refactoring: use mock contracts instead of real ones

### LINKED ISSUE #16: Use a mock environment to test SDK features.
Ok, then it is an issue of SDK tests. The unit tests should not rely on the deployed contract but they should use a mock-up. Perhaps we should introduce hardhat as a dev dependency and write SDK tests with that.

_Originally posted by @relu91 in https://github.com/vaimee/desmo-sdk/pull/14#discussion_r963474368_

### PR DESCRIPTION
**Pull request by iosonopersia**
Closes #16.

### CODE REVIEW

File: src/lib/utils/decoder.ts [hunk 1]
```diff
@@ -1,9 +1,3 @@
-/**
- * @file ./lib is a great place to keep all your code.
- * You can then choose what to make available by default by
- * exporting your lib modules from the ./src/index.ts entrypoint.
- */
-
```
Comment (relu91) [inline src/lib/utils/decoder.ts:6]: Unrelated change, why is it here? be careful next time.

Comment (iosonopersia) [reply src/lib/utils/decoder.ts:6]: Last 2 commits are unrelated,  I can push them after we merge this PR 👍

File: package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "build/types/**/*"
   ],
   "scripts": {
-    "test": "npm run build && mocha --inspect --require source-map-support/register --bail ./build/test/",
+    "test": "npm run build && mocha --inspect --require source-map-support/register --bail ./build/src/test/",
```
[the remark this answers is not in the document]
Comment (iosonopersia) [reply package.json]: Done.

**Comment by iosonopersia** [comment]
Please wait before merging, I'm trying to simplify things in the code...

**Review by relu91 — APPROVED** [review]
Let's go 👍🏻
